libfreerdp/primitives/neon/prim_YUV_neon.c

```diff
@@ -472,11 +472,11 @@ static pstatus_t neon_ChromaFilter(BYTE* WINPR_RESTRICT pDst[3], const UINT32 ds
 		BYTE* pU = pDst[1] + dstStep[1] * val2y;
 		BYTE* pV = pDst[2] + dstStep[2] * val2y;
 
-		if (val2y1 > nHeight)
+		if (val2y1 > nHeight + roi->top)
 			continue;
 
 		UINT32 x = roi->left / 2;
-		for (; x < halfWidth + roi->left / 2 - halfPad; x += 16)
+		for (; x < halfWidth + roi->left / 2 - halfPad; x += 8)
```

Comment:
sure it is only `+8`?
the data type holds 16 uint8_t and `vst2_u8` saves 16 uint8_t back

Reply:
I noticed that the index of pU/pV is twice x.
In these two lines inside the loop：
```
uint8x8x2_t u = vld2_u8(&pU[2 * x]); / uint8x8x2_t v = vld2_u8(&pV[2 * x]);
```